crates/uv-interpreter/src/python_query.rs

```diff
@@ -524,15 +535,68 @@ mod windows {
             return false;
         }
 
-        // Ex) `Local`
-        if !components
-            .next()
-            .is_some_and(|component| component.as_os_str() == "Local")
-        {
+        // The file is only relevant if it's a reparse point.
+        let Ok(md) = fs_err::symlink_metadata(path) else {
+            return false;
+        };
+        if md.file_attributes() & FILE_ATTRIBUTE_REPARSE_POINT == 0 {
             return false;
         }
 
-        true
+        let mut path_encoded = path
+            .as_os_str()
+            .encode_wide()
+            .chain(std::iter::once(0))
+            .collect::<Vec<_>>();
+
+        // SAFETY: The path is null-terminated.
+        #[allow(unsafe_code)]
+        let reparse_handle = unsafe {
+            CreateFileW(
+                path_encoded.as_mut_ptr(),
+                0,
+                0,
+                std::ptr::null_mut(),
+                OPEN_EXISTING,
+                FILE_FLAG_BACKUP_SEMANTICS | FILE_FLAG_OPEN_REPARSE_POINT,
+                std::ptr::null_mut(),
+            )
+        };
+
+        if reparse_handle == INVALID_HANDLE_VALUE {
+            return false;
+        }
+
+        let mut buf = [0u16; MAXIMUM_REPARSE_DATA_BUFFER_SIZE as usize];
+        let mut bytes_returned = 0;
+
+        // SAFETY: The buffer is large enough to hold the reparse point.
+        #[allow(unsafe_code, clippy::cast_possible_truncation)]
+        let success = unsafe {
+            DeviceIoControl(
+                reparse_handle,
+                FSCTL_GET_REPARSE_POINT,
+                std::ptr::null_mut(),
+                0,
+                buf.as_mut_ptr().cast(),
+                buf.len() as u32 * 2,
```

Comment:
Nit: not a 100 sure but I think we could use mem::size_of here (or size_of_value) to avoid the * 2